src/llmq/quorums.cpp

```diff
@@ -276,16 +276,13 @@ void CQuorumManager::EnsureQuorumConnections(Consensus::LLMQType llmqType, const
 
     bool allowWatch = gArgs.GetBoolArg("-watchquorums", DEFAULT_WATCH_QUORUMS);
     for (auto& quorum : lastQuorums) {
-        if (!CLLMQUtils::EnsureQuorumConnections(llmqType, quorum->pindexQuorum, myProTxHash, allowWatch)) {
+        if (CLLMQUtils::EnsureQuorumConnections(llmqType, quorum->pindexQuorum, myProTxHash, allowWatch)) {
```

Comment:
it removes gArgs.GetBoolArg.length - 2 calls by keeping it outside

Reply:
> Could move `bool allowWatch = gArgs.GetBoolArg("-watchquorums", DEFAULT_WATCH_QUORUMS);` into `CLLMQUtils::EnsureQuorumConnections` and drop `allowWatch` param.

> it removes gArgs.GetBoolArg.length calls (lastQuorums.length - 2 number of them) by keeping it outside

Thoughts about 2cbdacd92ab5f750e13e4ea84fd6b8108c889b32 697ed1e3dc0ae69216aaf8781d2d5fc837b09049 ?

> Also note that although results are the same `GetAllQuorumMembers` called inside of `CLLMQUtils::EnsureQuorumConnections` is much heavier than `IsMember` called here (esp. on mainnet). #4072 should fix this though :)

Good point! 👍 But yeah, should be fine with the cache i guess.

> Is there logic changes here or is it meant to be an optimization or done for readability purposes?

Yeah it's for readability only actually, because (at least to me) the two loops where the first removed the entries which should be skipped thing looked a bit confusing :)